src/ParamHandler.ts

```diff
@@ -104,7 +104,7 @@ export class ParamHandler {
     private parseValue(value: any, paramMetadata: ParamMetadata) {
         try {
             const parseValue = typeof value === "string" ? JSON.parse(value) : value;
-            if (paramMetadata.format !== Object && paramMetadata.format && this.driver.useClassTransformer) {
+            if (!(value instanceof paramMetadata.format) && paramMetadata.format !== Object && paramMetadata.format && this.driver.useClassTransformer) {
```

Comment:
The state field can be assigned to an object of a class that does not need to be converted.
An error occurs if you execute plainToClass on the object.